test/abstractarray.jl

```diff
@@ -263,27 +264,44 @@ function test_scalar_indexing{T}(::Type{T}, shape, ::Type{TestAbstractArray})
         end
     end
     # Test linear indexing and partial linear indexing
+    @test A == B
     i=0
     for i1 = 1:length(B)
         i += 1
         @test A[i1] == B[i1] == i
     end
-    i=0
-    for i2 = 1:size(B, 2)
-        for i1 = 1:size(B, 1)
-            i += 1
-            @test A[i1,i2] == B[i1,i2] == i
+    # TODO (#14770): run only for ndims(A) <= 2, elim. redirect_stderr stuff
+    if ndims(A) <= 2 || depwarns_not_error
+        filename = tempname()
+        open(filename, "w") do f
+            redirect_stderr(f) do
```

Comment:
Since the warning only gets issued on the first call, it would make this sensitive to which tests get assigned to which workers.